src/anemoi/models/models/encoder_processor_decoder.py

```diff
@@ -58,48 +63,50 @@ def __init__(
 
         self._define_tensor_sizes(config)
 
-        # Create trainable tensors
-        self._create_trainable_attributes()
-
         # Register lat/lon
-        self._register_latlon("data", self._graph_name_data)
-        self._register_latlon("hidden", self._graph_name_hidden)
+        for name in self._graph_mesh_names:
+            self._register_latlon(name)
 
         self.num_channels = config.model.num_channels
 
-        input_dim = self.multi_step * self.num_input_channels + self.latlons_data.shape[1] + self.trainable_data_size
+        input_dim = self.multi_step * self.num_input_channels
 
         # Encoder data -> hidden
-        self.encoder = instantiate(
-            config.model.encoder,
-            in_channels_src=input_dim,
-            in_channels_dst=self.latlons_hidden.shape[1] + self.trainable_hidden_size,
-            hidden_dim=self.num_channels,
-            sub_graph=self._graph_data[(self._graph_name_data, "to", self._graph_name_hidden)],
-            src_grid_size=self._data_grid_size,
-            dst_grid_size=self._hidden_grid_size,
-        )
+        self.encoders = nn.ModuleDict()
+        for data in self._graph_input_meshes:
```

Comment:
The variable should probably not be called `data`, as it seems this is a some sort of mesh name and not any data that is actually processed?